CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1.3)
+project(protobuf_root)
+include(flags OPTIONAL)
+if(DEFINED XP_NAMESPACE)
```

Comment:
We should be able to use xpro:: for now, and then in the future if we want to move to protobuf::, we can just remove XP_NAMESPACE from the project file, and then change our code